tools/serverpod_cli/lib/src/analyzer/dart/endpoint_analyzers/endpoint_class_analyzer.dart

```diff
@@ -32,15 +32,16 @@ abstract class EndpointClassAnalyzer {
 
   /// Returns true if the [ClassElement] is an endpoint class that should
   /// be validated and parsed.
-  static bool isEndpointClass(ClassElement element) {
-    if (element.supertype?.element.name != 'Endpoint') return false;
-    bool markedAsIgnored = element.metadata.any((annotation) {
-      var constant = annotation.computeConstantValue();
-      var type = constant?.type;
-      var typeName = type?.element?.name;
-      return typeName == ServerpodAnnotationClassNames.ignoreEndpoint;
-    });
-    if (markedAsIgnored) return false;
+  static bool isEndpointClass(
+    ClassElement element, {
+    bool respectIgnoreClassAnnotation = true,
+  }) {
+    if (element.allSupertypes.any((s) => s.element.name == 'Endpoint') !=
```

Comment:
This was like the existing logic, but ideally we'd probably check that `Endpoint` is from `serverpod/serverpod`. But I did not see a quick way to get the package path.